src/setting.rs

```diff
@@ -17,24 +17,29 @@
 //!
 //! The exact format still has to be reverse-engineered.
 
-use crate::util::nom_input_error_with_kind;
-use nom::error::ErrorKind;
-use nom::IResult;
+use binrw::{binread, NullString};
 
 #[derive(Debug)]
+#[binread]
+#[br(little)]
 /// Represents a setting file.
 pub struct Setting {
     /// Size of the string data field (should be always 96).
+    #[br(assert(len_stringdata == 0x60))]
     pub len_stringdata: u32,
     /// Name of the company ("PIONEER").
-    pub company: String,
+    #[br(pad_size_to = 0x20)]
+    pub company: NullString,
```

Comment:
```suggestion
    #[br(pad_size_to = 0x20)]
    #[brw(magic = b"PIONEER\x00")]
```
I'm actually not sure how `magic` works or is intended to be used, but I think we get writing for free.

Reply:
`magic` only works with fixed values. But This is sometimes `PIONEER`, sometimes `PIONEERDJ`.

Reply:
Thanks, this should be documented then I guess.

Reply:
I've seen these values:

```bash
$ for file in data/*/PIONEER/*.DAT; do cargo run --bin rekordcrate-setting -- "$file"; done | grep String
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/rekordcrate-setting data/demo_tracks/PIONEER/DEVSETTING.DAT`
    company: NullString("PIONEER DJ"),
    software: NullString("rekordbox"),
    version: NullString("6.6.1"),
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/rekordcrate-setting data/demo_tracks/PIONEER/DJMMYSETTING.DAT`
    company: NullString("PioneerDJ"),
    software: NullString("rekordbox"),
    version: NullString("1.000"),
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/rekordcrate-setting data/demo_tracks/PIONEER/MYSETTING2.DAT`
    company: NullString("PIONEER"),
    software: NullString("rekordbox"),
    version: NullString("0.001"),
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/rekordcrate-setting data/demo_tracks/PIONEER/MYSETTING.DAT`
    company: NullString("PIONEER"),
    software: NullString("rekordbox"),
    version: NullString("0.001"),
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/rekordcrate-setting data/empty/PIONEER/DEVSETTING.DAT`
    company: NullString("PIONEER DJ"),
    software: NullString("rekordbox"),
    version: NullString("6.6.1"),
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/rekordcrate-setting data/empty/PIONEER/DJMMYSETTING.DAT`
    company: NullString("PioneerDJ"),
    software: NullString("rekordbox"),
    version: NullString("1.000"),
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/rekordcrate-setting data/empty/PIONEER/MYSETTING2.DAT`
    company: NullString("PIONEER"),
    software: NullString("rekordbox"),
    version: NullString("0.001"),
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/rekordcrate-setting data/empty/PIONEER/MYSETTING.DAT`
    company: NullString("PIONEER"),
    software: NullString("rekordbox"),
    version: NullString("0.001"),
```

Reply:
https://github.com/Holzhaus/rekordcrate/pull/40/commits/d349f779c024897ed7eec8d0526bcd316ed090b1